extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentFinder.java

```diff
@@ -47,13 +46,13 @@
 {
   private static final Logger log = new Logger(S3DataSegmentFinder.class);
 
-  private final AmazonS3 s3Client;
+  private final EncryptingAmazonS3 s3Client;
   private final ObjectMapper jsonMapper;
   private final S3DataSegmentPusherConfig config;
 
   @Inject
   public S3DataSegmentFinder(
-      AmazonS3 s3Client,
+      EncryptingAmazonS3 s3Client,
```

Comment:
Instead of changing this everywhere, can you just bind `EncryptingAmazonS3` to `AmazonS3`? that way custom modules won't break

Reply:
That's a good point. I also considered that way, but chose the current implementation and added a javadoc for the reason.

```

Reply:
Our own interface with specific methods is fine.

Can `AmazonS3EncryptionClient` be used instead of wrapping all the requests?

Reply:
`AmazonS3EncryptionClient` is to perform client-side encryption. I renamed `EncryptingAmazonS3` to `ServerSideEncryptingAmazonS3`.